scrapy/utils/request.py

```diff
@@ -327,3 +327,35 @@ def _get_method(obj, name):
         return getattr(obj, name)
     except AttributeError:
         raise ValueError(f"Method {name!r} not found in: {obj}")
+
+
+def request_to_curl(request: Request) -> str:
+    """
+    Converts a :class:`~scrapy.Request` object to a curl command.
+
+    :param :class:`~scrapy.Request`: Request object to be converted
+    :return: string containing the curl command
+    """
+    method = request.method
+    data = (
+        "--data-raw '{}'".format(request.body.decode("utf-8")) if request.body else ""
+    )
+    headers = " ".join(
+        "-H '{}: {}'".format(k.decode(), v[0].decode())
+        for k, v in request.headers.items()
+    )
+    url = request.url
+    cookies = ""
+    if isinstance(request.cookies, dict) and request.cookies:
+        cookies = "--cookie '{}'".format(
+            "; ".join("{}={}".format(k, v) for k, v in request.cookies.items())
+        )
+    elif isinstance(request.cookies, list):
```

Comment:
Maybe this also needs a check that the list is not empty?

Reply:
Makes sense. The default is an empty dict, but I guess it is possible that an empty list ends up there.